LongWarOfTheChosen/Src/LW_WeaponsAndArmor/Classes/X2Item_PurifierWeapons.uc

```diff
@@ -0,0 +1,160 @@
+class X2Item_PurifierWeapons extends X2Item config(GameData_WeaponData);
+
+var config WeaponDamageValue ADV_PURIFIER_PISTOL_M1_WPN_BASEDAMAGE;
+var config WeaponDamageValue ADV_PURIFIER_PISTOL_M2_WPN_BASEDAMAGE;
+var config WeaponDamageValue ADV_PURIFIER_PISTOL_M3_WPN_BASEDAMAGE;
+
+static function array<X2DataTemplate> CreateTemplates()
+{
+    local array<X2DataTemplate> Weapons;
+    
+    Weapons.AddItem(CreateTemplate_AdvPurifierPistolM1_WPN());
+    Weapons.AddItem(CreateTemplate_AdvPurifierPistolM2_WPN());
+    Weapons.AddItem(CreateTemplate_AdvPurifierPistolM3_WPN());
+    return Weapons;
+}
+
+
+static function X2DataTemplate CreateTemplate_AdvPurifierPistolM1_WPN()
+{
+	local X2WeaponTemplate Template;
+
+	`CREATE_X2TEMPLATE(class'X2WeaponTemplate', Template, 'AdvPurifierPistolM1_WPN');
+	Template.WeaponPanelImage = "_Pistol";                       // used by the UI. Probably determines iconview of the weapon.
+	 
+	Template.ItemCat = 'weapon';
+	Template.WeaponCat = 'pistol';
+	Template.WeaponTech = 'magnetic';
+	Template.strImage = "";
+	Template.EquipSound = "Secondary_Weapon_Equip_Magnetic";
+
+	Template.RangeAccuracy = class'X2Item_SMGWeapon'.default.MIDSHORT_MAGNETIC_RANGE;
+	Template.BaseDamage = default.ADV_PURIFIER_PISTOL_M1_WPN_BASEDAMAGE;
+	Template.Aim = class'X2Item_DefaultWeapons'.default.PISTOL_MAGNETIC_AIM;
+	Template.CritChance = class'X2Item_DefaultWeapons'.default.PISTOL_MAGNETIC_CRITCHANCE;
+	Template.iClipSize = class'X2Item_DefaultWeapons'.default.PISTOL_MAGNETIC_ICLIPSIZE;
+	Template.iSoundRange = class'X2Item_DefaultWeapons'.default.PISTOL_MAGNETIC_ISOUNDRANGE;
+	Template.iEnvironmentDamage = class'X2Item_DefaultWeapons'.default.PISTOL_MAGNETIC_IENVIRONMENTDAMAGE;
+
+
+	Template.OverwatchActionPoint = class'X2CharacterTemplateManager'.default.PistolOverwatchReserveActionPoint;
+	Template.InfiniteAmmo = true;
+
+	Template.InventorySlot = eInvSlot_SecondaryWeapon;
+	Template.Abilities.AddItem('PistolStandardShot');
+	Template.Abilities.AddItem('PistolOverwatch');
+	Template.Abilities.AddItem('PistolOverwatchShot');
+	Template.Abilities.AddItem('HotLoadAmmo');
+	Template.Abilities.AddItem('Reload');
+
+	Template.SetAnimationNameForAbility('FanFire', 'FF_FireMultiShotMagA');
+	
+	// This all the resources; sounds, animations, models, physics, the works.
+	Template.GameArchetype = "WP_Pistol_MG.WP_Pistol_MG_Advent";
+
+	Template.iPhysicsImpulse = 5;
+
+	Template.CanBeBuilt = false;
+
+	Template.DamageTypeTemplateName = 'Projectile_MagAdvent';
+
+	Template.bHideClipSizeStat = true;
+
+	return Template;
```

Comment:
Isn't most of this the same for all the pistol variants? If so, it would be better to factor into a helper/utility function to do the basic template creation, and then each version-specific function can make the changes it needs.